luigi/scheduler.py

```diff
@@ -655,9 +672,15 @@ def add_task(self, task_id=None, status=PENDING, runnable=True,
         if deps is not None:
             task.deps = set(deps)
 
+        if deps_configs is not None:
+            task.deps_configs = deps_configs
+
         if new_deps is not None:
             task.deps.update(new_deps)
 
+        if new_deps_configs is not None:
+            task.deps_configs.update(new_deps_configs)
```

Comment:
Maybe think of it from the other side. Say you run TaskX(), which uses dynamic dependencies. You need `new_deps` because after each `yield`-from-a-`run()`, you will add a few more dependencies. However, for each `yield`-from-a-`run()` you don't update the retry-policy of the already added task TaskX(). Sure, the new deps laso have their own retry-policies but they are added in *their* `_add_task` call. Makes sense?